### PR TITLE
Add a Dev Console smoke test for Keycloak

### LINKED ISSUE #18853: Disabling Dev Services For Keycloak causes a template exception due to a strict rendering
### Describe the bug

Adding `quarkus.keycloak.devservices.enabled=false` to `application.proiperties` and starting with `quarkus:dev` and with `quarkus-oidc` included causes a template exception since the Dev UI templates shipped with `quarkus-oidc` check various properties created by the KC dev services processor which starts a container. I've spotted this problem after trying to accelerate the smote test (and not to start a container).

This is a problem since the plan is to also  have the custom non KC Dev UI supported OOB, #18427, which would require setting `quarkus.keycloak.devservices.enabled=false`. Also we will encourage to provide a custom Dev UI support for other providers.

I'll try to produce empty/null properties from the KC dev services processor if it is disabled. If that does not work then I believe the strict rendering for Dev UI should be switched off

CC @stuartwdouglas @mkouba @phillip-kruger 

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by sberyozkin**
Also
Fixes #18853.

(please backport together with #17364 if #17364 is backported)

### CODE REVIEW

File: extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/keycloak/KeycloakDevServicesProcessor.java [hunk 1]
```diff
@@ -215,7 +215,7 @@ private DevServicesConfigBuildItem prepareConfiguration(boolean createRealm,
     private StartResult startContainer() {
         if (!capturedDevServicesConfiguration.enabled) {
             // explicitly disabled
-            LOG.error("Not starting devservices for Keycloak as it has been disabled in the config");
+            LOG.debug("Not starting devservices for Keycloak as it has been disabled in the config");
```
Comment (sberyozkin) [inline extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/keycloak/KeycloakDevServicesProcessor.java:218]: It was `debug` in my original copy of a similar code but I changed it to `error` while I was looking at some PR problem and forgot to change it back

**Comment by sberyozkin** [comment]
Hmm, on Windows it tries to start a KC dev services container. I thought I disabled it, @mkouba, does the smoke test setup looks correct to you ?

**Comment by sberyozkin** [comment]
@mkouba I'm also not sure if I should go through all `provider.html` and also `embedded.html` and update all the references to various `info` properties with `??` even those inside the scripts/etc ? May be it is simpler to register them as `null` if `devservices` for KC are disabled ? It works though for now

**Review by phillip-kruger — APPROVED** [review]

**Comment by sberyozkin** [comment]
I've double checked it and on my laptop the container is not starting - though I had to update one unit test in `integration-tests/devmode` to ensure OIDC will not attempt to connect in a non-dev mode

**Comment by sberyozkin** [comment]
I've cancelled the build as I've realized I had to disable KC dev services for all the `devmode` tests - this is a problem with enabling by default - by at least there are only a few of these tests are there

**Comment by sberyozkin** [comment]
OK, it should be better now, I've disable it for all the tests at the pom level

**Comment by sberyozkin** [comment]
@phillip-kruger thanks, will also wait for Martin to check

**Comment by sberyozkin** [comment]
Looks like Martin is off, so I'll merge it then, I've verified locally as well that no container is started; the templates fixes are necessary in any case and we can do more as needed (I've run through the main page in a quickstart - all seems fine for now); Martin - please let me know when you get a chance if something else has to be improved :-)

**Pull request merged**